hudi-common/src/main/java/org/apache/hudi/common/table/timeline/TimelineUtils.java

```diff
@@ -188,20 +186,20 @@ public static List<String> getAffectedPartitions(HoodieTimeline timeline) {
           }
         case HoodieTimeline.CLEAN_ACTION:
           try {
-            HoodieCleanMetadata cleanMetadata = TimelineMetadataUtils.deserializeHoodieCleanMetadata(timeline.getInstantDetails(s).get());
+            HoodieCleanMetadata cleanMetadata = timeline.loadHoodieCleanMetadata(s);
             return cleanMetadata.getPartitionMetadata().keySet().stream();
           } catch (IOException e) {
             throw new HoodieIOException("Failed to get partitions cleaned at " + s, e);
           }
         case HoodieTimeline.ROLLBACK_ACTION:
           try {
-            return TimelineMetadataUtils.deserializeHoodieRollbackMetadata(timeline.getInstantDetails(s).get()).getPartitionMetadata().keySet().stream();
+            return timeline.loadHoodieRollbackMetadata(s).getPartitionMetadata().keySet().stream();
           } catch (IOException e) {
             throw new HoodieIOException("Failed to get partitions rolledback at " + s, e);
           }
         case HoodieTimeline.RESTORE_ACTION:
           try {
-            HoodieRestoreMetadata restoreMetadata = TimelineMetadataUtils.deserializeAvroMetadata(timeline.getInstantDetails(s).get(), HoodieRestoreMetadata.class);
+            HoodieRestoreMetadata restoreMetadata = timeline.loadInstantContent(s, HoodieRestoreMetadata.class);
```

Comment:
As discussed with @Davis-Zhang-Onehouse , we should have wrapper method in the timeline to read specific metadata and avoid calling `loadInstantContent(instant, clazz)` directly.